### PR TITLE
feat(INT-443): add initial support for execution chaining

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change
add support for chaining in default executor
Also refactor scenarios utils to accept scenarios in ts files.
Add support for direct use of node modules

### Development

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by koladilip — DISMISSED** [review]

**Review by mihir-4116 — DISMISSED** [review]

File: src/workflow/default_executor.ts [hunk 1]
```diff
@@ -15,18 +25,23 @@ export class DefaultWorkflowExecutor implements WorkflowExecutor {
       setContext: (key, value) => {
         context[key] = value;
       },
+      originalInput: input,
     };
 
-    let finalOutput: any;
+    let state: any = input;
```
Comment (saikumarrs) [inline src/workflow/default_executor.ts:31]: `curStepOutput` or `curOutput` sounds more appropriate than the generic name `state`, which doesn't offer anything by its face value.

Comment (saikumarrs) [reply src/workflow/default_executor.ts:31]: Also, it's better not to initialize it with the input value.

File: src/workflow/default_executor.ts [hunk 2]
```diff
@@ -15,18 +25,23 @@ export class DefaultWorkflowExecutor implements WorkflowExecutor {
...
-        const { skipped, output } = await stepExecutor.execute(input, executionBindings);
+        const { skipped, output } = await stepExecutor.execute(currentInput, executionBindings);
         if (skipped) {
           continue;
         }
         executionBindings.outputs[step.name] = output;
-        finalOutput = output;
+        state = output;
+        if (this.options.chainOutputs) {
+          currentInput = state;
```
Comment (saikumarrs) [inline src/workflow/default_executor.ts:43]: By replacing the value in `currentInput`, we lost access to the actual input to the step. Not that we need the input in the current implementation, but a slight alteration would improve the readability and cater to these needs in the future.

Something like this:

```
let curStepInput = input;
let prevStepOutput;
for (...) {
  if (chain) {
    curStepInput = prevStepOutput;
  }
  
  const curStepOutput = stepExecute(curStepInput);
  prevStepOutput = curStepOutput;
  // we can still make use of both the input and output of this step beyond this point
}
```

File: src/workflow/default_executor.ts [hunk 3]
```diff
@@ -54,3 +69,5 @@ export class DefaultWorkflowExecutor implements WorkflowExecutor {
     );
   }
 }
+
+export const chainExecutor = new DefaultWorkflowExecutor({ chainOutputs: true });
```
Comment (saikumarrs) [inline src/workflow/default_executor.ts:73]: Why are we exporting this default instance? It's not used anywhere.

**Review by saikumarrs — APPROVED** [review]

File: src/workflow/default_executor.ts (hunk 3 above)
Comment (koladilip) [reply src/workflow/default_executor.ts:73]: it is used in test cases

File: src/workflow/default_executor.ts (hunk 1 above)
Comment (koladilip) [reply src/workflow/default_executor.ts]: done

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**